packages/dedupe-mixin/README.md

```diff
@@ -0,0 +1,216 @@
++----+----+  +-----------+  +----------+
+     |
++----+----+
+| Logged  |
+| PageRed |
++---------+
+```
+
+```js
+class Page {}
+class PageRed extends Page {}
+class PageGreen extends Page {}
+class PageBlue extends Page {}
+class LoggedPagRed extends PageRed {}
+```
+
+If we want to start logging for PageGreen we have an issue:
+
+- we can't put the logic in `Page` as Blue should not be logged
+- we can't reuse the logic in `Logged PageGreen` as we can not extend from 2 source (even if we could it would mean conflicting info in Red and Green)
+
+What we can do is put it in an "external" place and write it so it can be "mixed in".
+
+```
+               +----------+                +----------+
+               |   Page   |                | Logging* |
+               +-+--+--+--+                +----------+
+                 |  |  |
+      +----------+  |  +-----------+
+      |             |              |
++-----+----+  +-----+-----+  +-----+----+
+| PageRed  |  | PageGreen |  | PageBlue |
+|  with    |  |   with    |  +----------+
+| Logging* |  |  Logging* |
++----------+  +-----------+
+```
+
+```js
+// defining the Mixin
+export const LoggingMixin = superclass =>
+  class LoggingMixin extends superclass {
+    // logging logic
+  };
+
+class Page {}
+// applying a Mixin
+class PageRed extends LoggingMixin(Page) {}
+class PageGreen extends LoggingMixin(Page) {}
+class PageBlue extends Page {}
+```
+
+With that approach we can extract logic into a separate code pieces we can use where needed.
+
+For a more in depth technical explanation please read [Real Mixins with JavaScript Classes](https://justinfagnani.com/2015/12/21/real-mixins-with-javascript-classes/).
+
+## Why is deduping of mixins necessary?
+
+We now want do logging for Red, Green and Blue.
+Easy enough - as we can now apply the LoggingMixin on the Page itself.
+
+```
+               +----------+               +----------+
+               |   Page   |               | Logging* |
+               |   with   |               +----------+
+               | Logging* |
+               +-+--+--+--+
+                 |  |  |
+      +----------+  |  +-----------+
+      |             |              |
++-----+----+  +-----+-----+  +-----+----+
+| PageRed  |  | PageGreen |  | PageBlue |
++----------+  |   with    |  +----------+
+              |  Logging* |
+              +-----------+
+```
+
+Oh oh - team Page was faster then Team Green. So the Mixin is now applied two times 😱
+So now our logging is completely wrong as suddenly Green does twice as many loggings.
+That is not what we intended 🙈
+
+What we need to do is make sure that each Mixin is attached only once even if we try to apply it multiple times.
+
+```js
+import { dedupeMixin } from '@open-wc/dedupe-mixin';
+
+export const MyMixin = dedupeMixin(
+  superclass =>
+    class MyMixin extends superclass {
+      // your mixin code goes here
+    },
+);
+```
+
+You can see exactly this situation in the demo.
```

Comment:
By applying `dedupeMixin` to the mixin function, before we export it, we can be sure that our mixin class will only take effect once, even if it is mixed in to multiple base classes in the inheritance chain.

Reply:
replace the sentence or add below?

Reply:
replace